packages/cozy-pouch-link/src/jsonapi.js

```diff
@@ -68,22 +87,23 @@ export const fromPouchResult = ({ res, withRows, doctype, client }) => {
   }
 
   if (withRows) {
-    const docs = res.rows
-      ? res.rows.map(row => row.doc).filter(filterDeletedDocumentsFromRows)
-      : res.docs
+    const docs = res.rows ? res.rows.map(row => row.doc) : res.docs
     const offset = res.offset || 0
+    normalizeDocs(client, doctype, docs)
 
-    return {
-      data: docs.map(doc => normalizeDoc(doc, doctype, client)),
+    const result = {
+      data: docs,
       meta: { count: docs.length },
       skip: offset,
       next: offset + docs.length < res.total_rows || docs.length >= res.limit
     }
+    return result
   } else {
+    Array.isArray(res)
+      ? normalizeDocs(client, doctype, res)
+      : normalizeDoc(client, doctype, res)
     return {
-      data: Array.isArray(res)
-        ? res.map(doc => normalizeDoc(doc, doctype, client))
-        : normalizeDoc(res, doctype, client)
+      data: res
```

Comment:
Did you check if we have unit tests that cover this `else` part? I'm not sure I did it before leaving last week.

Reply:
I added some tests here : 16c0f189250241f3c0fba6926c411ee7bd8b6b63